internal/cloud/aws.go

```diff
@@ -0,0 +1,126 @@
+/*
+Based on
+- https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+- https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html
+*/
+
+package cloud
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+const (
+	awsMetadataUrl      = "http://169.254.169.254/latest/"
```

Comment:
Isn't there any domain which we can point our requests too? Only IPs?

Reply:
From the documentation looks like there isn't. Blimey :-(

Reply:
No, there is not. It looks like something standard in cloud providers, as azure does it in the same way